platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -171,137 +230,269 @@ public Schema getSchema(CFMetaData metadata) {
+     * marked by columnToLoad
+     */
+    private static void loadColumns(SSTableReader ssTableReader, List<IAppendableColumn> columns,
+            PartitionColumns columnDefinitions, List<CassandraTokenRange> tokenRanges, String localEndpoint) {
+        int colCount = columnDefinitions.size();
+        ColumnFilter cf = ColumnFilter.selection(columnDefinitions);
+        List<Long> arrPrefixComparison = new ArrayList<>();
+        HashMap<CQL3Type, TypeSerializer<?>> serializers = new HashMap<CQL3Type, TypeSerializer<?>>();
+        Iterator<ColumnDefinition> colIter = columnDefinitions.regulars.simpleColumns();
+        // init reusable serializer and prefixComparison
+        while(colIter.hasNext()) {
+            ColumnDefinition cd = colIter.next();
+            CQL3Type type = cd.type.asCQL3Type();
+            serializers.put(type, type.getType().getSerializer());
+            arrPrefixComparison.add(cd.name.prefixComparison);
+        }
+
+        for (CassandraTokenRange tr : tokenRanges) {
+            // Load the partition if the local endpoint is listed as the first endpoint
+            if (tr.endpoints.get(0).equals(localEndpoint)) {
+                DataRange range = DataRange.forTokenRange(tr.tokenRange);
+                SSTableReadsListener listener = CassandraSSTableLoader.newReadCountUpdater();
+                ISSTableScanner currentScanner = ssTableReader.getScanner(cf, range, false, listener);
+
+                Spliterators.spliteratorUnknownSize(currentScanner, Spliterator.CONCURRENT)
+                    .forEachRemaining((partition) -> {
+                        Unfiltered unfiltered = partition.next();
+                        if (unfiltered instanceof Row) {
+                            Row row = (Row) unfiltered;
+                            int currentColumn = 0;
+                            Object value;
+                            IAppendableColumn col;
+                            // Extracting the value of each column from a single row
+                            for (ColumnData cd : row) {
+                                col = columns.get(currentColumn);
+                                // Missing column case #1 
+                                // Where unloaded missing column(s) is between idx 0 until last non missing column
+                                while (cd.column().name.prefixComparison != arrPrefixComparison
+                                        .get(currentColumn)) {
+                                    // The column with null value will not be included, so here 
+                                    // we add the null identifier for the skipped column(s)
+                                    col.appendMissing();
+                                    currentColumn++;
+                                    col = columns.get(currentColumn);
+                                }
+                                CQL3Type colType = cd.column().type.asCQL3Type();
```

Comment:
you don't need this, you already have them in the array you already built before.